app/src/pages/_app.tsx

```diff
@@ -1,45 +1,75 @@
-import { AppContext } from '@context/state'
 import { appWithTranslation } from 'next-i18next'
 import type { AppProps } from 'next/app'
-import useLocalStorage from 'src/hooks/useLocalStorage'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
 
 import Layout from '@components/Layout'
+import PageError from '@components/PageError'
 
+import useSessionStorage from '@src/hooks/useSessionStorage'
 import '@styles/styles.scss'
+import { getBackRoute, getForwardRoute, hasRoutingIssues } from '@utils/routing'
+import { initialSessionData } from '@utils/sessionData'
 
 function MyApp({ Component, pageProps }: AppProps) {
-  const [session, setSession] = useLocalStorage('session', {
-    clinic: undefined,
-    contact: {
-      firstName: '',
-      lastName: '',
-      phone: '',
-      comments: '',
-    },
-    eligibility: {
-      residential: '',
-      pregnant: false,
-      baby: false,
-      child: false,
-      guardian: false,
-      none: false,
-      loss: false,
-      before: '',
-      insurance: false,
-      snap: false,
-      tanf: false,
-      fdpir: false,
-      none2: false,
-    },
-  })
+  const sessionKey = 'session'
+  const [session, setSession] = useSessionStorage(
+    sessionKey,
+    initialSessionData
+  )
+
+  // Note: All router.push() calls have linting disabled on them.
+  // See https://nextjs.org/docs/api-reference/next/router#potential-solutions
+
+  // Pass in a prop for whether the form wizard is in review mode.
+  const router = useRouter()
+  const reviewMode = router.query.mode === 'review'
+
+  // Handle form wizard page access.
+  useEffect(() => {
+    try {
+      const outcome = hasRoutingIssues(router.pathname, session)
+      if (outcome.error) {
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        router.push({
+          pathname: outcome.cause,
+          query: { error: 'missing-data' },
+        })
+      }
```

Comment:
Notes for reviewers: This is where the gating/data validation per page is handled. If a user tries to access a page in the form wizard flow (say `/review`), but they haven't filled out valid data for all previous pages, then they will be routed to the "cause" (the first page in the flow that is missing data) and shown an error alert instructing them to fill out missing fields.

This handles two cases:
1. Link sharing: if a user is accidentally shared a link directly to a page in the middle of the form wizard flow, they will get directed back to the beginning.
2. Data corruption: if the data in the react state or session storage is corrupted somehow, the user flow can be rescued by sending them back to fill in the missing data

![Screen Shot 2022-10-28 at 2 49 26 AM](https://user-images.githubusercontent.com/67701/198558796-8a6d16c4-0650-431f-a5fb-2b22f2dba3bb.png)